runtime/dict.go

```diff
@@ -624,6 +624,34 @@ func dictKeys(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return toDictUnsafe(args[0]).Keys(f).ToObject(), nil
 }
 
+func dictPop(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	expectedTypes := []*Type{DictType, ObjectType, ObjectType}
+	argc := len(args)
+	if argc == 2 {
+		expectedTypes = expectedTypes[:2]
+	}
+	if raised := checkMethodArgs(f, "pop", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	key := args[1]
+	d := toDictUnsafe(args[0])
+	d.mutex.Lock(f)
+	item, raised := d.GetItem(f, key)
+	if raised == nil {
+		if item == nil {
+			if argc > 2 {
+				item = args[2]
+			} else {
+				raised = raiseKeyError(f, key)
+			}
+		} else {
+			_, raised = d.DelItem(f, key)
```

Comment:
When DelItem() raises, both item and raised will be non-nil when returned below. Some callers may assume that item != nil implies raised == nil.